internal/broker/broker.go

```diff
@@ -638,6 +631,37 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *session, au
 	return AuthGranted, userInfoMessage{UserInfo: authInfo.UserInfo}
 }
 
+// userNameIsAllowed checks whether the user's username is allowed to access the machine.
+func (b *Broker) userNameIsAllowed(userName string) bool {
+	// The user is allowed to login if:
+	// - ALL users are allowed
+	// - the user's name is in the list of allowed_users
+	// - the user is the owner of the machine
+	allowed := false
+	if b.cfg.userConfig.AllUsersAllowed() {
+		allowed = true
+	} else if b.cfg.userConfig.IsUserAllowed(userName) {
+		allowed = true
+	} else if b.cfg.userConfig.OwnerUserAllowed() {
+		// If owner is undefined, then the first user to login is considered the owner
+		if b.cfg.userConfig.OwnerIsUnset() || *b.cfg.userConfig.Owner() == userName {
+			allowed = true
+		}
+	}
+
+	// If the owner is unset and allowed, we auto-generate a config file with the first
+	// user to login as the owner.
+	if b.cfg.userConfig.OwnerUserAllowed() && b.cfg.userConfig.OwnerIsUnset() {
+		if b.cfg.ConfigFile != "" {
+			// TODO(nsklikas): What should we do in case of error?
```

Comment:
Opening a discussion here, since this should be handled.

Reply:
After our sync today, it was decided that authn should fail in this case. I will update it.